php/class-post-type.php

```diff
@@ -549,4 +548,37 @@ public function filter_user_has_cap( $allcaps, $caps ) {
 
 		return $allcaps;
 	}
+
+	/**
+	 * Publish snapshot changes when snapshot post is being published.
+	 *
+	 * @param int      $post_id Post id.
+	 * @param \WP_Post $post Post object.
+	 */
+	public function publish_snapshot( $post_id, $post ) {
```

Comment:
@PatelUtkarsh warning that when doing Publish in the Customizer, any existing snapshot will be transitioned to `publish`. See: https://github.com/xwp/wp-customize-snapshots/blob/f69b32592021cef7ffa5d34419f67f70950f7ec3/php/class-customize-snapshot-manager.php#L591

So it is important that this should short-circuit if `did_action( 'customize_save' )`, because this would indicate that the `customize_save` Ajax request is being made.